src/Kernel/IQSharpEngine.cs

```diff
@@ -345,12 +365,16 @@ public override async Task<ExecutionResult> ExecuteMundane(string input, IChanne
             {
                 try
                 {
-                    await this.Initialized;
-                    // Once the engine is initialized, we know that Workspace
+                    // Since this method is only called once this.Initialized
+                    // has completed, we know that Workspace
                     // and Snippets are both not-null.
+                    Debug.Assert(
+                        this.Initialized.IsCompleted,
+                        "Engine was not initialized before call to ExecuteMundane. " +
+                        "This is an internal error; if you observe this message, please file a bug report at https://github.com/microsoft/iqsharp/issues/new."
```

Comment:
I really like this idea about asking the user to file an issue if they reach an Assert!